jerry-libc/include/assert.h

```diff
@@ -25,6 +25,11 @@ extern "C"
 #endif /* __cplusplus */
 
 #ifndef NDEBUG
+
+#ifndef __GNUC__
+#define __builtin_expect(expression, expected_value) (expression)
+#endif /* !__GNUC__ */
```

Comment:
@tilmannOSG && @zherczeg -- I'd like to propose a different  way of looking at this:

The GCC constructs _are_ useful (to GCC :-); removing them seems like we're getting rid of functionality.  Since, with this #def, they behave just fine for every other compiler, why not leave them?

Secondly, "unlikely" is only defined in jrt.h; using it all over the place forces (read: makes it a really good idea :-) us to reorganize things in .h files (for this change, because now we suddenly have to include jrt.h in assert.h, which makes no obvious sense) -- these kinds of reorganizations are required maintenance for an evolving project, but kind of out of the scope for this otherwise simple change (since the bigger fix requires system architectural knowledge, discussion of division, what/when/where to include what, etc... :-).

...so for the record: I'm not trying to get out of work by suggesting that we leave this code as written :-)...but doing this correctly will take a lot of work (and reduce functionality!), so I _am_ actually saying that we should check this in as is...